examples/local_module.py

```diff
@@ -6,3 +6,4 @@
 """
 
 N = 100
+N
```

Comment:
this example is actually not executed by SG (it does not match the build pattern) so this shouldn't do anything. Can you modify an example that runs, e.g., https://1042-25860190-gh.circle-artifacts.com/0/rtd_html/auto_examples/plot_function_identifier.html you could write `font_size` at the end and we should see it (if you also update our `doc/conf.py` to match, which would be worthwhile)

Reply:
Yes, I can do that. Just to confirm I can change our `doc/conf.py` `print_eval_repr` to `('__repr__',)`?

Also, as you suggested it would be nice to give an example of this in action like a pandas dataframe - but would I have to import this if I were to use it? I could instead define my own class like in the test and use this to produce some html - maybe in a separate gallery as I am not sure it fits into any of the current galleries.  I would also have to change the config for our documentation to `('_repr_html_', '__repr__')` (which still might work fine for the other galleries, I will check).

I will also update the documentation to talk about capturing the other repr's 

Finally, I just realised that we don't `eval()` anything now and maybe we should change the name of this config. Maybe `'capture_repr'` or something similar...?

Reply:
Sure `capture_repr` seems good

Yes you can change `doc/conf.py`. I think it's also fine to add a pandas dependency for the doc build for the purposes of showing the `__html_repr__` (eventually) and/or `__repr__` (in this PR). Basically we want to show a use case that will be common, and maybe that will be the most common one.

Reply:
But this PR will now be able to show `_repr_html_` now?

Reply:
In that case I'd change our `doc/conf.py` to have `__repr_html__` and `__repr__` in it, demo both in some example, and also update the docs in this PR to mention `__repr_html__` since it's not in there currently

Reply:
Excellent.

Sorry, can you add pandas as dependency for just the doc build or do you have to add it for the whole package?

Reply:
Just for the doc build is fine. Add it here:

https://github.com/sphinx-gallery/sphinx-gallery/blob/master/.circleci/config.yml#L29

And in `intersphinx` here:

https://github.com/sphinx-gallery/sphinx-gallery/blob/master/doc/conf.py#L312

as `'pandas': ('https://pandas.pydata.org/pandas-docs/stable', None),`

Reply:
... and looking at our existing examples, making a dedicated example for this probably makes sense. You can show that `__html_repr__` and `__repr__` both work, depending on what the object actually exposes. And that assigments do not print.